HydroModelBuilder/ModelInterface/flopyInterface/viz/MT3D_PP_viz.py

```diff
@@ -256,3 +267,155 @@ def viewConcsByZone(self, nper='all', specimen=None):
 
     plt.show()
 # End viewConcsByZone()
+
+
+def _plot_obs_vs_sim(self, obs_set, obs_sim_zone_all, unc=None):
+    scatterx = np.array([h[0] for h in obs_sim_zone_all])
+    scattery = np.array([h[1] for h in obs_sim_zone_all])
+
+    residuals = [loc[0] - loc[1] for loc in obs_sim_zone_all]
+
+    # First step is to set up the plot
+    width = 20
+    height = 5
+    multiplier = 1.
+    fig = plt.figure(figsize=(width * multiplier, height * multiplier))
+
+    ax = fig.add_subplot(1, 2, 1)  # , aspect='equal')
+    ax.set_title('Residuals')
+
+    #colours = ['r', 'orangered', 'y', 'green', 'teal', 'blue', 'fuchsia']
+    ax.hist(residuals, bins=20, alpha=0.5, color='black', histtype='step', label='all')
+
+    plt.legend(loc='upper left', ncol=4, fontsize=11)
+
+    ax = fig.add_subplot(1, 2, 2)
+    ax.set_title('{}: Sim vs Obs ({} points)'.format(obs_set.upper(), len(scatterx)))
+
+    ax.scatter(scatterx, scattery, facecolors='none', alpha=0.5)
+
+    plt.xlabel('Observed')
+    plt.ylabel('Simulated', labelpad=10)
+
+    sum1 = 0.0
+    sum2 = 0.0
+
+    if len(scatterx) != 0:
+        mean = np.mean(scatterx)
+        for i in range(len(scatterx)):
+            num1 = (scatterx[i] - scattery[i])
+            num2 = (scatterx[i] - mean)
+            sum1 += num1**np.float64(2.0)
+            sum2 += num2**np.float64(2.0)
+
+        ME = 1 - sum1 / sum2
+
+        ymin, ymax = ax.get_ylim()
+        xmin, xmax = ax.get_xlim()
+
+        ax.text(xmin + 0.45 * (xmax - xmin), ymin + 0.4 * (ymax - ymin),
+                'Model Efficiency = %4.2f' % (ME))
+
+        # for PBIAS
+        def pbias(simulated, observed):
```

Comment:
These functions should be moved out as I've noticed the same function definition gets reused elsewhere.